Jenkinsfile

```diff
@@ -149,9 +149,16 @@ pipeline {
                                 def packageFiles = findFiles(glob: '**/package.json')
                                 for (int i = 0; i < packageFiles.size(); i++) {
                                     dir(packageFiles[i].path.split('package.json')[0]) {
-                                        echo "Scanning ${packageFiles[i].name}"
-                                        nodejs(configId: 'npmrc-default', nodeJSInstallationName: 'nodejs') {
-                                            sh 'nsp check'
+                                        def packageFile = readJSON file: 'package.json'
+                                        if (packageFile.scripts =~ /.*webpack.*/ || packageFile.containsKey("browserify")) {
+                                            nodejs(configId: 'npmrc-default', nodeJSInstallationName: 'nodejs') {
+                                                try {
+                                                    echo "Scanning ${packageFiles[i].path}"
+                                                    sh 'nsp check'
+                                                } catch (err) {
+                                                    echo "Failed for ${packageFiles[i].path}"
```

Comment:
If we catch the error will this still fail the build?

Reply:
It will not fail the build this way. The reason I'm proposing we add this is because nsp does not currently seem to have a way to suppress vulnerabilities. The only vulnerabilities that are being reported at the moment are from a transitive that should actually have been listed as a non-production dependency. Without a way to suppress this there is no way to gracefully handle incorrectly declared production dependencies inherited transitively

Reply:
Strike that, it actually describes how to do it in the nsp tools readme, there were just no docs on the website... I'll remove the catch statement and instead add an exception for the library in question

Reply:
We can add suppressions via an .nsprc file for the module in question. We currently do this for the catalog-ui-search and admin ui. https://github.com/nodesecurity/nsp

Reply:
yep, just found that when I double checked 👍